book/src/guides/indent.md

```diff
@@ -46,6 +46,19 @@ capture on the same line, the indent level isn't changed at all.
 - `@outdent` (default scope `all`):
 Decrease the indent level by 1. The same rules as for `@indent` apply.
 
+- `@extend-indented`:
+Extend the range of this node to the end of the line and to lines that
+are indented more than the line that this node starts on. This is useful
+for languages like Python, where for the purpose of indentation some nodes
+(like functions or classes) should also contain indented lines that follow them.
+
+- `@stop-extend`:
+Prevents the first extension of an ancestor of this node. For example, in Python
+a return expression always ends the block that it is in. Note that this only prevents
+the next extension of one ancestor: If multiple ancestors can be extended (for example
+multiple nested conditional blocks in python), only the extension of the innermost
+ancestor is prevented.
```

Comment:
I think the names could use some workshopping - I like "extend" but `@stop-extend` doesn't convey well that it only stops the extension once

Reply:
Maybe `@indent.extend` and then `@indent.extend.pause`? Or `@extend` and `@extend.pause-once`? I also liked "after" from the discussion in the issue

Reply:
`@extend` sounds good. In case we add more "extend" captures types, we can still rename it to something more specific later. Maybe `@extend.prevent-once` is a more accurate description of what the other capture does? Or `@extend.end-once`?

Reply:
Just a question as a reader, why is it called `@extend.prevent-once`? More specifically, I'm not sure I understand how it is semantically "once". If I understand correctly, the capture type is supposed to convey that the extended range of the most immediate ancestor ends. Would `@extend.end-immediate-ancestor` be more apt albeit verbose?

Reply:
I think the wording in the docs might be a bit confusing: It prevents the first extension of an ancestor, but this ancestor doesn't necessarily have to be immediate. Rather, this refers to the first ancestor _that is captured by an `@extend` capture_.

The last sentence in the docs ("All other ancestors are unaffected (regardless of whether the innermost ancestor would actually have been extended)") could probably be explained a bit better. What it means is that the first ancestor that is captured by an `@extend` may not actually cause any extension (for example because the lines after it are not indented enough). In this case, `@extend.prevent-once` "prevents" this extension, but since the extension wouldn't have happened anyways, it has no effect. In particular, if some more distant ancestor is actually extended, this extension is _not_ prevented.

For python, these subtleties don't really make a difference but I still think it's important to document them, since for other languages they might be important. If you find a better name than `@extend.prevent-once` or a better wording for the docs, feel free to open a PR.

Reply:
Ah thank you for the explanation! So technically, we can say the most immediate ancestor captured by `@extend` capture.

Would it be easier to simply think of `@extend` and `@extend.prevent-once` as opposite operations, like opening and closing parentheses? We could then rename them into `@extend.begin-once` and `@extend.end-once`, or `@extend.open` and `@extend.close` to emphasize this; something with symmetry in the naming. Is this a correct characterization?

Reply:
`@extend` and `@extend.prevent-once` are sort of opposites but it's not completely symmetric. With parentheses, there's usually exactly one closing parenthesis for each opening parenthesis. Here, this is not the case: There can be multiple `@extend` captures without any `@extend.prevent-once` (in which case the innermost one takes precedence) and there might be multiple `@extend.prevent-once` captures without any `@extend` (although they have no effect in this situation). Also, the `@extend.prevent-once` doesn't appear _after_ the `@extend` but rather as a descendant of it, in other words, it is _contained_ in it.

I'm not sure if there is a better name for this, but the documentation could definitely be improved. Maybe a link to the python indent queries could also be helpful as an example for how to use the captures in practice.